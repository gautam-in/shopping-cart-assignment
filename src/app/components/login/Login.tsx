import React, { useEffect, useState } from "react";
import { useRouter } from "next/router";
import { useSelector, useDispatch } from "react-redux";
import { authAction, isUserLoggedInState } from "../../../store/auth-slice";

const Login = () => {
  const [userObj, setUserObj] = useState({ email: "", password: "" });
  const [loginState, setLoginState] = useState(
    useSelector(isUserLoggedInState)
  );
  const dispatcher = useDispatch();

  console.log(" loginState ", loginState);
  const fetcher = (param: any) =>
    fetch(param.url, {
      method: "POST",
      headers: { "Content-Type": "apllication/json" },
      body: JSON.stringify({
        email: param.userObj.email,
        password: param.userObj.password,
      }),
    })
      .then((res) => {
        console.log("res1 ", res);
        return res.json();
      })
      .then((data) => {
        console.log(" data:", data);
        if (data.message) {
          setLoginState(false);
          dispatcher(authAction.setLoginState(false));
        } else if (data.id) {
          dispatcher(authAction.setLoginState(true));
          setLoginState(true);
        }

        return data;
      })
      .catch((error) => {
        console.error("Error:", error);
        setLoginState(false);
        dispatcher(authAction.setLoginState(false));
      });

  const loginHandler = (event: any) => {
    event.preventDefault();
    const { email, password } = document.forms[0];
    console.log("loginHandler: ", email.value, password.value);
    if (userObj) {
      const param = { url: `/api/auth/login`, userObj: userObj };
      fetcher(param);
    }
    setUserObj({ email: email.value, password: password.value });
  };
  const router = useRouter();

  if (loginState === true) {
    console.log("loginstate : - ", loginState);
    router.push("/offers");
  } else {
    // router.push("/login");
  }

  return (
    <div className="px-40 py-28 flex flex-wrap justify-center ">
      <form onSubmit={loginHandler}>
        <div className="lg:mr-10 sm:mb-10 truncate ">
          <h1 className="font-bold text-2xl mb-2">Login</h1>
          <p className="text-gray-600 text-sm text-center">
            Get access to your Orders, Wishlists and Reccomendations
          </p>
        </div>
        <div className="">
          <p className="text-green-500">Email</p>
          <input
            type="text"
            name="email"
            className="border-b-2 w-full"
            placeholder="enter email"
          />
          <input
            type="password"
            name="password"
            className="border-b-2 w-full mb-10"
            placeholder="password"
          />

          <input
            className="block bg-red-800 text-white w-full"
            type="submit"
            name="login"
            value="Login"
          />
        </div>
      </form>
    </div>
  );
};

export default Login;
