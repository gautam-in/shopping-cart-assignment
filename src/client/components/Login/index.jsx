import React, { useState, useCallback } from 'react';
import { useHistory } from 'react-router';

import formHook from '../../hooks/formHook';
import WEB_PATH from '../../constants/webPath';
import { EMAIL, PASSWORD } from '../../constants/regex';

import LoginPage from './LoginPage';

export const Login = () => {
  const history = useHistory();
  const [fields, setFields] = useState({
    email: {
      isRequired: true,
      requiredError: 'Please enter the email.',
      value: '',
      isMatching: EMAIL,
      matchError: 'Invalid Email',
      error: '',
    },
    password: {
      isRequired: true,
      requiredError: 'Please enter the password.',
      value: '',
      isMatching: PASSWORD,
      matchError: 'Invalid password.',
      error: '',
    },
  });

  const {
    validateInputs,
    onInputChange,
    onSubmitValidate,
    formActions,
  } = formHook({ setFields });

  const onSubmit = useCallback((e, values) => {
    e.preventDefault();
    const error = onSubmitValidate(values);
    if (!error) {
      history.push(WEB_PATH.HOME);
    }
  }, [validateInputs]);

  return (
    <LoginPage
      fields={fields}
      formActions={formActions}
      onInputChange={onInputChange}
      onSubmit={onSubmit}
    />
  );
};

export default {
  component: Login,
};
