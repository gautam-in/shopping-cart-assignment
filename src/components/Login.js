import React, { Component } from "react";
import { Link } from "react-router-dom";
import { toast } from "react-toastify";
import { emailValidation } from "../commonData/index";

class Login extends Component {
  constructor(props) {
    super(props);

    this.state = {
      email: "",
      password: "",
      emailValidation: "",
      buttonEnableValidation: true,
      errorMessageOnScreen: "",
    };
  }

  validation = (data, id) => {
    if (id === "email") {
      if (emailValidation(data)) {
        this.setState({
          emailValidation: false,
        });
      } else {
        this.setState({
          emailValidation: true,
        });
      }
    }
  };

  onChangeInputType = (event) => {
    this.setState({
      [event.target.id]: event.target.value,
    });
    this.validation(event.target.value, event.target.id);
    setTimeout(() => {
      this.buttonEnableValidation();
    }, 0);
  };

  handleSubmit = (event) => {
    event.preventDefault();
    let tempEmail = this.state.email;
    let tempPass = this.state.password;
    let userExist = sessionStorage.getItem(tempEmail);
    if (userExist !== null) {
      if (tempPass === userExist) {
        // console.log("Success");
        sessionStorage.setItem("currentUser", this.state.email);
        this.props.history.push("/products");
      } else {
        // console.log("InCorrect Password");
        this.setState({
          errorMessageOnScreen: "InCorrect Password, Please Try Again!",
        });
        toast.warn("InCorrect Password, Please Try Again!");
      }
    } else {
      // console.log("User does not exist");
      this.setState({
        errorMessageOnScreen:
          "User does not exist, Please Click here to Create an Account!",
        email: "",
        password: "",
      });
      toast.error("User does not exist, Please Create an Account");
    }
  };

  buttonEnableValidation = () => {
    if (
      this.state.email &&
      this.state.password &&
      !this.state.emailValidation
    ) {
      this.setState({
        buttonEnableValidation: false,
      });
    } else {
      this.setState({
        buttonEnableValidation: true,
      });
    }
  };

  render() {
    return (
      <>
        <section className="loginContainer">
          <article className="loginLeftSide">
            <h1 className="loginTitle">Login</h1>
            <p className="loginPara">
              Get Access to your Orders, Wishlist and Recommendations
            </p>
          </article>
          <article className="loginRightSide">
            <form autoComplete="off">
              {/* <label htmlFor="firstName">First Name</label> */}
              <input
                type="email"
                id="email"
                value={this.state.email}
                className="inputType"
                placeholder="Email"
                onChange={this.onChangeInputType}
                style={{
                  borderBottom: this.state.emailValidation
                    ? "red solid 2px"
                    : "rgb(177, 175, 175) solid 2px",
                }}
              />
              <p className="inputtypeError">
                {this.state.emailValidation
                  ? "Please add correct Email/Gmail Address"
                  : ""}
              </p>
              {/* <label htmlFor="lastName">Last Name</label> */}
              <input
                type="password"
                id="password"
                value={this.state.password}
                className="inputType"
                placeholder="Password"
                onChange={this.onChangeInputType}
              />
              <button
                type="submit"
                className={
                  this.state.buttonEnableValidation
                    ? "submitButtonDisabled"
                    : "submitButton"
                }
                onClick={
                  this.state.buttonEnableValidation
                    ? function () {}
                    : this.handleSubmit
                }
                title={
                  this.state.buttonEnableValidation
                    ? "Please Fill all the Fields"
                    : "Click here to Login"
                }
              >
                Login
              </button>
              <Link
                to="/signup"
                style={{
                  padding: "2px 0px ",
                  marginTop: "2px",
                  color: "blue",
                  fontWeight: "bold",
                  display: "block",
                  textAlign: "right",
                  fontSize: "11px",
                }}
              >
                If you don't have account, Create new one!
              </Link>
              <Link
                to={
                  this.state.errorMessageOnScreen ===
                  "InCorrect Password, Please Try Again!"
                    ? "/signin"
                    : "/signup"
                }
                style={{
                  padding: "2px 0px ",
                  marginTop: "6px",
                  color: "red",
                  fontWeight: "bold",
                  display: "block",
                }}
              >
                {this.state.errorMessageOnScreen
                  ? this.state.errorMessageOnScreen
                  : ""}
              </Link>
            </form>
          </article>
        </section>
      </>
    );
  }
}

export default Login;
