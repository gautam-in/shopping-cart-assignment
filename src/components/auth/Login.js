import React from "react";
import "./styles.css";
import TextInput from "../textInput/TextInput";
import { emailValidation, password } from "../utils/validationRules";
import { useForm } from "react-hook-form";
import { useSelector } from "react-redux";
import { useHistory } from "react-router-dom";
export default function Login() {
  const {
    register,
    handleSubmit,
    formState: { errors },
  } = useForm();

  let authenticated = false;
  const signUpDetail = useSelector((state) => state.getAuthDetail.signUp);
  const history = useHistory();
  const onSubmit = (data) => {
    signUpDetail.forEach((detail) => {
      if (detail.email === data.email && detail.password === data.password) {
        history.push("/product");
        authenticated = true;
      }
    });
    if (!authenticated) {
      alert("Kindly signup before you Login");
    }
  };
  return (
    <div className="app-login">
      <div className="left-block">
        <h2>Login</h2>
        <p>Get acess to orders , Wishlist and Recommendations.</p>
      </div>
      <form onSubmit={handleSubmit(onSubmit)} className="right-block">
        <TextInput
          type="text"
          name="email"
          rules={emailValidation}
          register={register}
          errors={errors}
          label="Email"
        />
        <TextInput
          type="password"
          name="password"
          label="Password"
          rules={password}
          register={register}
          errors={errors}
        />
        <button className="app-btn"> Login</button>
      </form>
    </div>
  );
}
