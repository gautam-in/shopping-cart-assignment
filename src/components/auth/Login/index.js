import React, { useState } from 'react'
import FloatingLabelInput from '../../../common/FloatingLabelInput'
import { language } from '../../../Utils'
// import '../auth.scss'

function Login() {
    const [Email, setEmail] = useState('');
    const [Password, setPassword] = useState('');
    return (
        <div className='auth-component'>
            <div className="main pg-lr-sp">
                <section className='section1'>
                    <div>
                        <h2>Login</h2>
                        <p>{language.login_message}</p>
                    </div>
                </section>
                <section className='section2'>
                        <FloatingLabelInput
                            className="FloatingLabelInput"
                            label={"Email"}
                            onChange={(e) => { setEmail(e.target.value) }}
                            value={Email}
                        />
                        <FloatingLabelInput
                            className="FloatingLabelInput"
                            label={"Password"}
                            onChange={(e) => { setPassword(e.target.value) }}
                            value={Password}
                        />
                        <button className='butt'>Login</button>
                    
                </section>
            </div>
        </div>
    )
}

export default Login

