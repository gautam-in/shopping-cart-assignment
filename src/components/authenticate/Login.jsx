import React from "react";
import "./styles.css";
import TextInput from "../inputField/TextInput";
import { useForm } from "react-hook-form";
import { useSelector } from "react-redux";
import { useHistory } from "react-router-dom";
import { emailValidation, password } from "../utils/common";

export default function Login() {
  const {
    register,
    handleSubmit,
    formState: { errors },
  } = useForm();

  let authenticated = false;
  const signUpDetails = useSelector((state) => state.getAuthenticateDetails.signUp);
  const history = useHistory();
  const onSubmit = (data) => {
    signUpDetails.forEach((detail) => {
      if (detail.email === data.email && detail.password === data.password) {
        history.push("/product");
        authenticated = true;
      }
    });
    if (!authenticated) {
      alert("Please signup before you Login to the app");
    }
  };
  return (
    <div className="app-login">
      <div className="login-left">
        <h2>Login</h2>
        <p>Get acess to your Orders ,Wishlist and Recommendations</p>
      </div>
      <form onSubmit={handleSubmit(onSubmit)} className="login-right">
        <TextInput
          type="text"
          name="email"
          rules={emailValidation}
          register={register}
          errors={errors}
          label="Email"/>
        <TextInput
          type="password"
          name="password"
          label="Password"
          rules={password}
          register={register}
          errors={errors}/>
        <button className="app-btn"> Login</button>
      </form>
    </div>
  );
}