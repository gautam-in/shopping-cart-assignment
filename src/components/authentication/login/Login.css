.login-wrapper {
    flex-direction: column;
    align-items: center;
    justify-content: center;
    gap: 1rem;
    padding: 2rem 0;

}

.login-form-wrapper, .login-header-wrapper {
    flex-direction: column;
    gap: 1.5rem;
    align-items: center;
}
.login-header-wrapper {
    align-items: center;
}
.login-header-wrapper > p{
    text-align: center;
}
.login-form-wrapper > label > p {
    margin-bottom: 0.3rem;
    transition: ease 0.5s;
}
.login-form-btn {
    background-color: #D90467;
    color: whitesmoke;
    border: 0;
}
.login-form-input {
    border: 0;
    outline: 0;
    border-bottom: 1px solid grey;
    width: 15rem;
    padding: 0.5rem 1rem 0.2rem 0.3rem;
    font-size: 1rem;
}
.login-input {
    text-align: center;
}
.login-input :first-child {
    position: relative;
    top: 0.5rem;
    text-align: left;
}
.login-input:focus-within > p:first-child {
    color: #4ED9D9;
    top: -0.5rem;
}
.login-input:focus-within > input {
    border-bottom: 2px solid #4ED9D9;
}
.error-message {
    color: red;
    margin-top: 0.5rem;
    text-align: center;
}
.show-password {
    position: absolute;
    right: 5%;
    top: 55%;
}
@media only screen and (min-width: 768px){
    .login-wrapper {
        flex-direction: row;
        align-items: flex-start;
        gap: 4rem;
        padding: 3rem;
    }

    .login-form-input {
        width: 20rem;
    }
    .btn {
        padding: 1rem;
    }
    .login-form-wrapper {
        gap: 3rem;
    }
}
@media only screen and (min-width: 912px){
    .login-form-input {
        width: 25rem;
    }
}