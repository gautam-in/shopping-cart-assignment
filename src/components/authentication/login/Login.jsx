import React, { useState } from 'react'
import  "./Login.css"
import { useNavigate } from 'react-router-dom'
import { validateLoginForm } from './validateLoginForm.jsx'

const Login = () => {

  const [formData, setFormaData] = useState({email:"", password:""})
  const [error, setError] = useState({})

  const navigate = useNavigate()

  const loginClickHandler = () => {
    const formError = validateLoginForm(formData)
    if(Object.keys(formError).length < 1){
      setError(() => ({}))
      navigate("/")
    }
    else {
      setError(() => formError)
    }
  }
  
  return (
    <main className = "flex login-wrapper bg-color" data-testid = "login-form">
        <div className = "flex login-header-wrapper">
          <h2>Login</h2>
          <p>Get access to your Orders, Wishlist and Recommendations</p>
        </div>
        <form className = "flex login-form-wrapper" onSubmit={(e) => e.preventDefault()}>
            <p>All fields marked with an asterisk(*) are required</p>
            <label htmlFor = "email" className = "login-input">
                <p>Email <span aria-hidden = "true"className = "error-message">*</span></p>
                <input type = "email" id = "email" data-testid = "email" className = "login-form-input" onChange = {(e) => setFormaData((prev) => ({...prev, email:e.target.value}))} required/>
                {error.email && <p className = "error-message" data-testid = "email-error"><span className = "warning-sign">&#9888;</span> {error.email}</p>}

            </label>
            
            <label htmlFor = "password" className = "login-input">
                <p>Password<span aria-hidden = "true" className = "error-message">*</span></p>
                <input type = "password" id = "password" data-testid = "password" className = "login-form-input" onChange = {(e) => setFormaData((prev) => ({...prev, password:e.target.value}))} required/>
                {error.pwd && <p className = "error-message" data-testid = "pwd-error"><span className = "warning-sign">&#9888;</span> {error.pwd}</p>}
            </label>
            <button className = "btn login-form-btn" onClick = { loginClickHandler}>Login</button>
        </form>
    </main>
  )
}

export default Login