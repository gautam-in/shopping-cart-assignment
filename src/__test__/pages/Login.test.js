import React from "react";
import { shallow, configure } from 'enzyme';
import Login from "../../components/pages/Login";
import Adapter from 'enzyme-adapter-react-16';
jest.mock('../../components/LoginComponent');

configure({ adapter: new Adapter() })

const wrapper = shallow(<Login />);
let container, containerProp, childContainer, childContainerProps;

describe("Login", () => {
  beforeEach(() => {
    container = wrapper.find("div");
    containerProp = container.props();
  });

  it('should render our Snapshots correctly', () => {
    const wrapper = shallow(<Login />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should have a <div>", () => {
     expect(container).toHaveLength(1);
  });

  describe("LoginComponent", () => {
    beforeEach(() => {
      childContainer = wrapper.find('LoginComponent');
      childContainerProps = childContainer.props();
    });

    it("should have a <LoginComponent>", () => {
      expect(childContainer).toHaveLength(1);
    });
  });
});