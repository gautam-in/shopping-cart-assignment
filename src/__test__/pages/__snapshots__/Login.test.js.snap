// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render our Snapshots correctly 1`] = `ShallowWrapper {}`;
